src/index.cpp

```diff
@@ -958,7 +963,7 @@ void Index::index_field_in_memory(const field& afield, std::vector<index_record>
             });
         } else if(afield.is_array()) {
             // handle vector index first
-            if(afield.type == field_types::FLOAT_ARRAY && afield.num_dim > 0) {
+            if(afield.num_dim > 0 && (afield.type == field_types::FLOAT_ARRAY || afield.type == field_types::BOOL_ARRAY)) {
```

Comment:
Let's add a `afield.is_vector_type()` helper so we can reuse it in other places.